app/models/comment.rb

```diff
@@ -202,9 +210,14 @@ def wrap_timestamps_if_video_present!
   def shorten_urls!
     doc = Nokogiri::HTML.fragment(processed_html)
     doc.css("a").each do |anchor|
-      unless anchor.to_s.include?("<img") || anchor.to_s.include?("<del") || anchor.attr("class")&.include?("ltag")
-        anchor.content = strip_url(anchor.content) unless anchor.to_s.include?("<img") # rubocop:disable Style/SoleNestedConditional
+      next if anchor.inner_html.include?("<img")
+
+      urls = anchor.content.scan(URI_REGEXP).flatten
+      anchor_content = anchor.content
+      urls.each do |url|
+        anchor_content.sub!(/#{Regexp.escape(url)}/, strip_url(url))
       end
+      anchor.inner_html = anchor.inner_html.sub!(/#{Regexp.escape(anchor.content)}/, anchor_content)
```

Comment:
this looks like it should only be done if the sub! returns a non-empty response (don't try to set inner_html to nil)

https://gist.github.com/djuber/45829ea7c5160347dedf83141b124cd7

Reply:
Oops, that is one I did not see coming. Sorry about the bug :sweat_smile: